### PR TITLE
Add fill_value option on stack methods to replace invalid values

### PR DESCRIPTION
**Pull request by QRemy**
Add a fill_value option on .to_masked() and .stack() methods to replace non-finite values as suggested in :
https://github.com/gammapy/gammapy/pull/3416#discussion_r665947289

### CODE REVIEW

File: gammapy/maps/hpxnd.py [hunk 1]
```diff
@@ -382,8 +384,9 @@ def stack(self, other, weights=None):
                 "Can only stack equivalent maps or cutout of the same map."
             )
 
-        data = other.quantity.to_value(self.unit)
-
+        data = other.quantity.to_value(self.unit).copy()
```
Comment (adonath) [inline gammapy/maps/hpxnd.py:387]: Could avoid the copy here, if `nan_to_num = False`...

**Review by adonath — APPROVED** [review]
Thanks @QRemy, I have left one minor inline comment otherwise looks fine!

File: gammapy/maps/hpxnd.py (hunk 1 above)
Comment (QRemy) [reply gammapy/maps/hpxnd.py:387]: Without the copy I have two tests failing with a message about "data is read only"

Comment (adonath) [reply gammapy/maps/hpxnd.py:387]: I meant moving the `.copy()` into the `if` block...

**Review by adonath — APPROVED** [review]
Thanks @QRemy, no further comments from my side...

**Pull request merged**